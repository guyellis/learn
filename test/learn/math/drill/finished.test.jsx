const constants = require('../../../../src/learn/common/constants');
const React = require('react');
const Finished = require('../../../../src/learn/math/drill/finished');
const renderer = require('react-test-renderer');
const MuiThemeProvider = require('material-ui/styles/MuiThemeProvider').default;
const getMuiTheme = require('material-ui/styles/getMuiTheme').default;
const lightBaseTheme = require('material-ui/styles/baseThemes/lightBaseTheme').default;

const {
  RECORD_NEW,
  // RECORD_EQUAL,
  // RECORD_MISS,
  // RECORD_NOT_EXIST,
} = constants;
const muiTheme = getMuiTheme(lightBaseTheme);

describe('Finished', () => {
  test('should be rendered a new record', () => {
    const previousResults = [{
      actual: 5,
      id: 1,
      task: [5, 5, 0, 10],
      timeTaken: 2.2,
    }];
    const resultInfo = {
      text: 'Result Info Text',
      newRecordInfo: RECORD_NEW,
    };

    const component = renderer.create(
      <MuiThemeProvider muiTheme={muiTheme}>
        <Finished
          levelIndex={0}
          opIndexes={[0]}
          previousResults={previousResults}
          resultInfo={resultInfo}
          timeAllowed={600}
          timeLeft={100}
          totalProblems={20}
        />
      </MuiThemeProvider>,
    );

    const finishedBadge = component.toJSON();
    expect(finishedBadge).toMatchSnapshot();
  });
});
