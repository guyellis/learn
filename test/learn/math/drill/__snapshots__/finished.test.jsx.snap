// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Finished should be rendered a new record 1`] = `
<div>
  <h1>
    Finished
  </h1>
  <div
    style={
      Object {
        "backgroundColor": "#8bc78b",
        "borderRadius": "5px",
        "color": "black",
        "marginBottom": "10px",
        "padding": "20px",
      }
    }
  >
    Result Info Text
  </div>
  <div
    style={Object {}}
  >
    You need 10 or more correct answers to earn a badge. On this 
test you got 0 answer(s) correct.
  </div>
  <div>
    You had 100 seconds left out of the 600 seconds allowed.
  </div>
  <div>
    You correctly answered 0 of the 20 problems.
  </div>
  <h4>
    The problem that took you the longest to answer
  </h4>
  <div>
    <div
      style={
        Object {
          "color": "red",
          "lineHeight": "35px",
        }
      }
    >
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        +
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        =
      </span>
      <span
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": 20,
            "padding": 3,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        ✗
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        2.2 seconds
      </span>
    </div>
  </div>
  <h4>
    The problem you answered the fastest
  </h4>
  <div>
    <div
      style={
        Object {
          "color": "red",
          "lineHeight": "35px",
        }
      }
    >
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        +
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        =
      </span>
      <span
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": 20,
            "padding": 3,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        ✗
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        2.2 seconds
      </span>
    </div>
  </div>
  <div>
    <h4>
      The ones that you got wrong
    </h4>
    <div>
      <div
        style={
          Object {
            "color": "red",
            "lineHeight": "35px",
          }
        }
      >
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          5
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          +
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          5
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          =
        </span>
        <span
          style={
            Object {
              "borderStyle": "solid",
              "borderWidth": 1,
              "marginLeft": 20,
              "padding": 3,
            }
          }
        >
          5
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          ✗
        </span>
        <span
          style={
            Object {
              "paddingLeft": 20,
            }
          }
        >
          2.2 seconds
        </span>
      </div>
    </div>
  </div>
  <h4>
    How you did (sorted by slowest)
  </h4>
  <div>
    <div
      style={
        Object {
          "color": "red",
          "lineHeight": "35px",
        }
      }
    >
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        +
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        =
      </span>
      <span
        style={
          Object {
            "borderStyle": "solid",
            "borderWidth": 1,
            "marginLeft": 20,
            "padding": 3,
          }
        }
      >
        5
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        ✗
      </span>
      <span
        style={
          Object {
            "paddingLeft": 20,
          }
        }
      >
        2.2 seconds
      </span>
    </div>
  </div>
</div>
`;
